src/Resources/config/services/factory.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <defaults autowire="false" autoconfigure="false" public="true" />
+
+        <service id="Sylius\RefundPlugin\Factory\CreditMemoSequenceFactory" />
+
+        <service id="Sylius\RefundPlugin\Factory\RefundPaymentFactory">
+            <argument type="service" id="sylius.repository.payment_method"/>
+        </service>
+
+        <service
+            id="sylius_refund.custom_factory.credit_memo"
```

Comment:
What do you think about setting id to `Sylius\RefundPlugin\Factory\CreditMemoFactory`? The current name is not used in any place outside of this configuration and will not be public. This way we won't have to call it a "custom factory".